insecure/cmd/corrupted_builder.go

```diff
@@ -0,0 +1,57 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/insecure/corruptible"
+	"github.com/onflow/flow-go/module"
+)
+
+// CorruptibleConduitFactoryPort is the port number that gRPC server of the conduit factory of corrupted nodes is listening on.
+const CorruptibleConduitFactoryPort = "4300"
```

Comment:
What is this different port used for? Can you maybe explain at a high level how the corruptible conduit system works?

I was not a reviewer on the previous PR's, but based on what I'm seeing in this PR, here is my approximate understanding:
Basically, we have a gRPC server running on node A that allows us to manually send messages to it, which then get delivered to the node via the corrupted conduit as if it had actually arrived from another node B on the network?
Are these "corrupted" messages interleaved with the "real" messages received from node B? Is there a mechanism that enables us to drop certain "real" messages sent by node B?

Btw, not sure if you have seen this, but libp2p has developed [TestGround](https://github.com/testground/testground) which they used to do testing for the libp2p implementation. It may be useful to look into this for inspiration.

Reply:
The port is used for communicating between the "orchestrator" and each corrupted node, on their own private network. To find a detailed explanation of the design search for "Corruptible Conduits Framework" in Nortion.

Corrupted nodes communicate on their own "attack network" that's outside of the regular protocol libp2p network.

Corrupted messages are not interleaved real messages. Corrupted messages are delivered to the orchestrator which then decides what to do with them - drop them, alter them, or pass them through unaltered. The orchestrator sends the messages it receives back to the conduit which then sends them on the network. Each corrupted conduit is "dumb" and only sends its messages to the orchestrator and then sends whatever the orchestrator sends it back. If the orchestrator wants to drop a message, it doesn't send anything back to the corrupted conduit and the message never gets sent.